test/LaunchDarkly.EventSource.Tests/EventSourceEndToEndTest.cs

```diff
@@ -29,100 +29,78 @@ public void ReceiveEventStreamInChunks()
+                    var expectedActions = new List<EventSink.Action>();
+                    expectedActions.Add(EventSink.OpenedAction());
+                    foreach (var data in eventData)
+                    {
+                        expectedActions.Add(EventSink.MessageReceivedAction(new MessageEvent(MessageEvent.DefaultName, data, uri)));
+                    }
+
+                    var config = Configuration.Builder(uri).LogAdapter(_testLogging).Build();
+                    using (var es = new EventSource(config))
+                    {
+                        var sink = new EventSink(es);
+                        _ = es.StartAsync();
+                        sink.ExpectActions(expectedActions.ToArray());
+                    }
                 }
             }
-        }
-
-        [Fact]
-        public void ReadTimeoutIsDetectedInDefaultStringStreamReadingMode()
-        {
-            TimeSpan readTimeout = TimeSpan.FromMilliseconds(200);
-            using (var server = StartWebServerOnAvailablePort(out var uri, ctx =>
+            finally
             {
-                ctx.Response.ContentType = "text/event-stream";
-                ctx.Response.SendChunked = true;
-                var stream = ctx.Response.OutputStream;
-                WriteChunk(stream, "data: event1\n\ndata: e");
-                Thread.Sleep(readTimeout + readTimeout);
-                WriteChunk(stream, "vent2\n\n");
-            }))
-            {
-                var config = Configuration.Builder(uri).LogAdapter(_testLogging)
-                    .ReadTimeout(readTimeout).Build();
-                using (var es = new EventSource(config))
-                {
-                    var sink = new EventSink(es) { Output = _testLogger.Debug };
-                    _ = es.StartAsync();
-                    sink.ExpectActions(
-                        EventSink.OpenedAction(),
-                        EventSink.MessageReceivedAction(new MessageEvent(MessageEvent.DefaultName, "event1", uri)),
-                        EventSink.ErrorAction(new ReadTimeoutException()),
-                        EventSink.ClosedAction()
-                        );
-                }
+                allEventsReceived.Set();
             }
         }
 
-        [Fact]
-        public void ReadTimeoutIsDetectedInRawUtf8ReadingMode()
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void ReadTimeoutIsDetected(bool utf8Mode)
         {
             TimeSpan readTimeout = TimeSpan.FromMilliseconds(200);
-            using (var server = StartWebServerOnAvailablePort(out var uri, ctx =>
+            IEnumerable<string> DoChunks()
             {
-                ctx.Response.ContentType = "text/event-stream";
-                ctx.Response.SendChunked = true;
-                var stream = ctx.Response.OutputStream;
-                WriteChunk(stream, "data: event1\n\ndata: e");
+                yield return "";
```

Comment:
This is a bit weird but I found that if I did not write a zero-length chunk at the start of the stream like this, `Stream.ReadAsync` would not pick up the next non-zero-length chunk right away but only when the next one had been written— as if the first one had not been flushed— which would mess up any tests that involved timeouts. This would only happen in Windows, and I couldn't find any documented reason for it; there doesn't seem to be any such requirement for chunked HTTP responses, in fact a zero-length chunk isn't even supposed to be a thing in chunked HTTP. But it does consistently make a difference. I'm not too concerned about the mysteriousness of this because we are verifying in many other tests that the client code is able to read chunked streams correctly; it's only this fake server whose behavior is weird, and that's not part of the library.